crates/rpc/rpc-eth-api/src/helpers/state.rs

```diff
@@ -90,19 +93,11 @@ pub trait EthState: LoadState + SpawnBlocking {
         let chain_info = self.chain_info()?;
         let block_id = block_id.unwrap_or_default();
 
-        // if we are trying to create a proof for the latest block, but have a BlockId as input
-        // that is not BlockNumberOrTag::Latest, then we need to figure out whether or not the
-        // BlockId corresponds to the latest block
-        let is_latest_block = match block_id {
-            BlockId::Number(BlockNumberOrTag::Number(num)) => num == chain_info.best_number,
-            BlockId::Hash(hash) => hash == chain_info.best_hash.into(),
-            BlockId::Number(BlockNumberOrTag::Latest) => true,
-            _ => false,
-        };
-
-        // TODO: remove when HistoricalStateProviderRef::proof is implemented
-        if !is_latest_block {
-            return Err(EthApiError::InvalidBlockRange)
+        // Check whether the distance to the block exceeds the maximum configured lookback.
+        let block_number = self.provider().block_number_for_id(block_id)?;
```

Comment:
this should return a block not found error, no reason to continue if we can't find the block here